packages/grafana-data/src/types/pluginExtensions.ts

```diff
@@ -39,6 +39,8 @@ export type PluginExtensionComponent<Props = {}> = PluginExtensionBase & {
   component: React.ComponentType<Props>;
 };
 
+export type ComponentTypeWithMeta<Props = {}> = React.ComponentType<Props> & { meta: PluginExtensionComponentMeta };
```

Comment:
Hmm, I know too long names are not good either, but maybe `ExtensionsComponentTypeWithMeta` or `ExtensionComponentWithMeta` would be more specific? 

In this case I would also think about adding a short comment explaining the difference between this and the `PluginExtensionComponent` type above, as I think that can be a bit confusing? E.g.:

```ts
// `ComponentTypeWithMeta` is the type of React components that we actually return to the 
// extension-points (not what we store in the extension registry and not what plugins register).
```

Reply:
Great feedback! I will update it 👍🏻